sql/index_builder.go

```diff
@@ -285,9 +285,19 @@ func (b *IndexBuilder) updateCol(ctx *Context, colExpr string, potentialRanges .
 		return
 	}
 
+	if b.idx.IsSpatial() {
+		if _, ok := currentRanges[0].LowerBound.(BelowNull); ok {
```

Comment:
This method is supposed to intersect two arbitrary nary sets of current and new ranges. This addition appears to intersect single dimension ranges for one specific edge case: 1) when there is only one current and potential range expression, 2) and we can take the current range lower and new range upper bound. I don't think this is correct, can we make it do a general intersection, or not use lookup builder for spatial indexes if this logic is correct and they are subject to very different range intersection rules?